agent/src/main/java/com/microsoft/hydralab/agent/runner/xctest/XctestRunner.java

```diff
@@ -0,0 +1,99 @@
+import com.microsoft.hydralab.common.util.Const;
+import com.microsoft.hydralab.common.util.FileUtil;
+import com.microsoft.hydralab.common.util.ShellUtils;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+public class XctestRunner extends TestRunner {
+    private static String zipPath = "";
+    private static String folderPath = "";
+    private static String resultPath = "";
+
+    public XctestRunner(DeviceManager deviceManager, TestTaskRunCallback testTaskRunCallback) {
+        super(deviceManager, testTaskRunCallback);
+    }
+
+    @Override
+    protected void run(DeviceInfo deviceInfo, TestTask testTask, TestRun testRun) throws Exception {
+        Logger reportLogger = testRun.getLogger();
+        ScreenRecorder deviceScreenRecorder = deviceManager.getScreenRecorder(deviceInfo, testRun.getResultFolder(), reportLogger);
+        deviceScreenRecorder.setupDevice();
+        deviceScreenRecorder.startRecord(testTask.getTimeOutSecond());
+        // Need a location here
+        unzipXctestFolder(testTask.getAppFile(), testRun, reportLogger);
+        runXctest();
+        deviceScreenRecorder.finishRecording();
+        analysisXctestResult();
+
+    }
+
+    private void unzipXctestFolder(File zipFile, TestRun testRun, Logger reportLogger) {
+        reportLogger.info("start unzipping file");
+        zipPath = zipFile.getAbsolutePath() + "/" + Const.SmartTestConfig.ZIP_FOLDER_NAME + "/";
+        folderPath = testRun.getResultFolder().getAbsolutePath() + "/" + Const.SmartTestConfig.XCTEST_ZIP_FOLDER_NAME
+                + "/";
+        resultPath = testRun.getResultFolder().getAbsolutePath();
+
+        FileUtil.unzipFile(zipFile.getAbsolutePath(), folderPath);
+    }
+
+    private String runXctest(DeviceInfo deviceInfo, String scope, String suiteName,
+                             String testPkgName, String testRunnerName, Logger logger,
+                             IShellOutputReceiver receiver, TestRun testRun,
+                             int testTimeOutSec, Map<String, String> instrumentationArgs) {
+        if (deviceInfo == null) {
+            throw new RuntimeException("No such device: " + deviceInfo);
+        }
+        if (testTimeOutSec <= 0) {
+            // the test should not last longer than
+            testTimeOutSec = 45 * 60;
+        }
+        StringBuilder argString = new StringBuilder();
+        if (instrumentationArgs != null && !instrumentationArgs.isEmpty()) {
+            instrumentationArgs.forEach((k, v) -> argString.append(" ").append(k).append(" ").append(v));
+        }
+        String commFormat;
+        if (StringUtils.isBlank(argString.toString())) {
+            commFormat = "xcodebuild test-without-building";
+        } else {
+            commFormat = "xcodebuild test-without-building" + argString;
```

Comment:
What's the dependency of this command? Any environmental requirement changes please clarify it in the agent set up guide: https://github.com/microsoft/HydraLab/wiki/Deploy-a-test-agent-service#ios-mac-agent

Reply:
Ok, I will update the doc to declare the dependencies.